index.js

```diff
@@ -1,30 +1,162 @@
-main(input, 30);
+
+const consume = async (msgHandler, errorHandler) => {
+    let message = await getMessage();
+    if (!message) {
+        console.log('No messages in queue');
+        consume(msgHandler, errorHandler);
+    } else {
+        try {
+            await msgHandler(message);
+        } catch (err) {
+            errorHandler(err);
+        }
+        consume(msgHandler, errorHandler);
+    }
+};
+
+const messageHandler = async (msg) => {
+    if (!isValidMessage(msg)) throw new Error('Invalid message from queue');
```

Comment:
nit: create constants for these errors, and add detail, e.g, `msg`, `prompt` to the error itself for debugging. otherwise, you know there is error, but you don't know what causes the error

in order for make error aggregate-able, perhaps define your own error class for example, this way, you can COUNT(1) GROUP BY message to measure the volume, but use detail to debug
```
class CustomError extends Error {
    constructor(message, detail) {
        super(message);
        this.detail = detail;
    }
}
```

If you want to seperate retriable and non-retriable error to use DLQ or not, then you can add a `errorCode` field to it, and then conditionally send to DLQ

Reply:
I really love this suggestion, super valuable. I think the constants and including more detail are great takes but creating a custom error class is really great. Thanks I will implement it